compiler/qsc_frontend/src/compile.rs

```diff
@@ -94,50 +93,51 @@ impl SourceMap {
         entry: Option<Arc<str>>,
     ) -> Self {
         let mut offset_sources = Vec::new();
-        for (name, contents) in sources {
-            offset_sources.push(Source {
-                name,
-                contents,
-                offset: next_offset(&offset_sources),
-            });
-        }
 
         let entry_source = entry.map(|contents| Source {
```

Comment:
The entry expression (if it exists) now comes first in the source map. This way we can append more sources to the end without the offsets conflicting with the entry expression's offset.